File: app/src/main/java/com/aleksandra/go4mytrip/PackingListAdapter.java
package com.aleksandra.go4mytrip;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.localbroadcastmanager.content.LocalBroadcastManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class PackingListAdapter extends RecyclerView.Adapter<PackingListAdapter.MyViewHolder> {
    private Context context;
    private List<PackingModel> mData;
    private static int TYPE_PACK = 1;
    private static int TYPE_SHOP = 2;
    Boolean isPacking;

    public PackingListAdapter(Context context, List<PackingModel> mData, Boolean isPacking) {
        this.context = context;
        this.mData = mData;
        this.isPacking = isPacking;

    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view;
        if (viewType == TYPE_PACK) {
            LayoutInflater layoutInflater = LayoutInflater.from(context);
            view = layoutInflater.inflate(R.layout.item_packinglist, parent, false);
            return new MyViewHolder(view);
        } else {
            LayoutInflater layoutInflater = LayoutInflater.from(context);
            view = layoutInflater.inflate(R.layout.item_shopping_list, parent, false);
            return new MyViewHolder(view);
        }
    }

    @Override
    public int getItemViewType(int position) {
        if (isPacking) {
            return TYPE_PACK;

        } else {
            return TYPE_SHOP;
        }
    }

    @Override
    public void onBindViewHolder(@NonNull final MyViewHolder holder, final int position) {
        holder.name.setText(mData.get(position).getName());
        holder.checkBox.setChecked(mData.get(position).getChecked());
        holder.sCheckBox.setChecked(mData.get(position).getCheckedS());
        holder.addToShoppingList.setChecked(mData.get(position).getToBuy());


        holder.addToShoppingList.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Boolean isPressed;
//                Boolean isPressed = mData.get(position).getToBuy();
//                isPressed = !isPressed;

                if (holder.addToShoppingList.isChecked()) {
                    isPressed = true;
                } else {
                    isPressed = false;
                }


                String id = mData.get(position).getItemId();
                Intent intentToBuy = new Intent("toBuy");
                intentToBuy.putExtra("isPressed", isPressed);
                intentToBuy.putExtra("idPckItem", id);
                LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentToBuy);
            }

        });

        holder.deletePackingItem.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //packingListListener.onClickDelete(mData.get(position), position);
                String id = mData.get(position).getItemId();

                Intent intentPackingItemDelete = new Intent("delete-item");
                intentPackingItemDelete.putExtra("idPackingItemDelete", id);
                LocalBroadcastManager.getInstance(context).sendBroadcast(intentPackingItemDelete);
            }
        });


        holder.checkBox.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Boolean isCheck;
                String id = mData.get(position).getItemId();
                String name = mData.get(position).getName();
                String category = mData.get(position).getCategory();
                if (holder.checkBox.isChecked()) {
                    isCheck = true;
                } else {
                    isCheck = false;
                }

                Intent intentCheckBox = new Intent("checkbox");
                intentCheckBox.putExtra("isCheck", isCheck);
                intentCheckBox.putExtra("idPckItem", id);
                intentCheckBox.putExtra("name", name);
                intentCheckBox.putExtra("category", category);
                LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentCheckBox);

            }
        });
        holder.sCheckBox.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Boolean isCheckS;
                String id = mData.get(position).getItemId();

                if (holder.sCheckBox.isChecked()) {
                    isCheckS = true;
                    // holder.name.setPaintFlags(Paint.STRIKE_THRU_TEXT_FLAG);
                } else {
                    isCheckS = false;
                    // holder.name.setPaintFlags(holder.name.getPaintFlags() & (~Paint.STRIKE_THRU_TEXT_FLAG));
                }


                Intent intentCheckBoxS = new Intent("checkboxS");
                intentCheckBoxS.putExtra("isCheckS", isCheckS);
                intentCheckBoxS.putExtra("idPckItem", id);
                LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentCheckBoxS);

            }
        });

    }

    //    public String getItem(int position) {
//        return mData.get(position).getItemId();
//    }
    @Override
    public int getItemCount() {
        return mData.size();
    }


    public static class MyViewHolder extends RecyclerView.ViewHolder {
        TextView name;
        CheckBox sCheckBox;
        CheckBox checkBox;
        ImageView deletePackingItem;
        CheckBox addToShoppingList;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            name = itemView.findViewById(R.id.text_item);
            checkBox = itemView.findViewById(R.id.checkBox_item);
            sCheckBox = itemView.findViewById(R.id.sCheckBox_item);
            deletePackingItem = itemView.findViewById(R.id.deletePackingItem);
            addToShoppingList = itemView.findViewById(R.id.addToShoppingList);

        }
    }
}

File: app/src/main/java/com/aleksandra/go4mytrip/lists/PackingListAdapter.kt
package com.aleksandra.go4mytrip.lists

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.ImageView
import android.widget.TextView
import androidx.localbroadcastmanager.content.LocalBroadcastManager
import androidx.recyclerview.widget.RecyclerView
import com.aleksandra.go4mytrip.R

class PackingListAdapter(private val context: Context, private val mData: MutableList<PackingModel>, private var isPacking: Boolean) : RecyclerView.Adapter<PackingListAdapter.MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view: View
        return if (viewType == TYPE_PACK) {
            val layoutInflater = LayoutInflater.from(context)
            view = layoutInflater.inflate(R.layout.item_packinglist, parent, false)
            MyViewHolder(view)
        } else {
            val layoutInflater = LayoutInflater.from(context)
            view = layoutInflater.inflate(R.layout.item_shopping_list, parent, false)
            MyViewHolder(view)
        }
    }

    override fun getItemViewType(position: Int): Int {
        return if (isPacking) {
            TYPE_PACK
        } else {
            TYPE_SHOP
        }
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.name.text = mData[position].name
        holder.checkBox.isChecked = mData[position].checked!!
        holder.sCheckBox.isChecked = mData[position].checkedS!!
        holder.addToShoppingList.isChecked = mData[position].toBuy!!

        holder.addToShoppingList.setOnClickListener {
            val id = mData[position].itemId
            val name = mData[position].name
            val category = mData[position].category
            val isCheck = holder.checkBox.isChecked
            val isCheckedS = holder.sCheckBox.isChecked
            val toBuy =  holder.addToShoppingList.isChecked
            val intentToBuy = Intent(context, ShoppingList::class.java)
            val packingModel = PackingModel(id, name, category, isCheckedS, isCheck, toBuy)
            intentToBuy.putExtra(PackingList.TOBUY, packingModel)
            intentToBuy.action = "toBuy"
            LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentToBuy)
        }
        holder.deletePackingItem.setOnClickListener {
            val id = mData[position].itemId
            val name = mData[position].name
            val category = mData[position].category
            val isCheck = holder.checkBox.isChecked
            val isCheckedS = holder.sCheckBox.isChecked
            val toBuy =  holder.addToShoppingList.isChecked
            val packingModel = PackingModel(id, name, category, isCheckedS, isCheck, toBuy)
            val intentPackingItemDelete = Intent(context, PackingList::class.java)
            intentPackingItemDelete.putExtra(PackingList.DELETE, packingModel)
            intentPackingItemDelete.action = "delete-item"
            LocalBroadcastManager.getInstance(context).sendBroadcast(intentPackingItemDelete)
        }
        holder.checkBox.setOnClickListener {
            val id = mData[position].itemId
            val name = mData[position].name
            val category = mData[position].category
            val isCheck = holder.checkBox.isChecked
            val isCheckedS = holder.sCheckBox.isChecked
            val toBuy =  holder.addToShoppingList.isChecked
            val packingModel = PackingModel(id, name, category, isCheckedS, isCheck, toBuy)
            val intentCheckBox = Intent(context, PackingList::class.java)
            intentCheckBox.putExtra(PackingList.CHECKBOX, packingModel)
            intentCheckBox.action = "checkbox"
            LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentCheckBox)
        }
        holder.sCheckBox.setOnClickListener {
            val id = mData[position].itemId
            val name = mData[position].name
            val category = mData[position].category
            val isCheck = holder.checkBox.isChecked
            val isCheckedS: Boolean = holder.sCheckBox.isChecked
            val toBuy =  holder.addToShoppingList.isChecked
            val packingModel = PackingModel(id, name, category, isCheckedS, isCheck, toBuy)
            val intentCheckBoxS = Intent(context, ShoppingList::class.java)
            intentCheckBoxS.putExtra(ShoppingList.CHECKBOXS, packingModel)
            intentCheckBoxS.action = "checkboxS"
            LocalBroadcastManager.getInstance(context).sendBroadcastSync(intentCheckBoxS)
        }
    }

    override fun getItemCount(): Int {
        return mData.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var name: TextView
        var sCheckBox: CheckBox
        var checkBox: CheckBox
        var deletePackingItem: ImageView
        var addToShoppingList: CheckBox

        init {
            name = itemView.findViewById(R.id.text_item)
            checkBox = itemView.findViewById(R.id.checkBox_item)
            sCheckBox = itemView.findViewById(R.id.sCheckBox_item)
            deletePackingItem = itemView.findViewById(R.id.deletePackingItem)
            addToShoppingList = itemView.findViewById(R.id.addToShoppingList)
        }
    }

    companion object {
        private const val TYPE_PACK = 1
        private const val TYPE_SHOP = 2
    }
}